test/util/payload-test.js

```diff
@@ -50,12 +51,14 @@ function buildTest ({ testFile, context, dir, file, webhook }) {
         id = t.context.id
       }
 
-      const endpoint = buildEndpoint({ id, type, httpServerPort })
+      // TODO: use ID for future /:id hooks or remove all references entirely
+      const endpoint = buildEndpoint({ type, httpServerPort })
+      const signatureBuilder = getBuildSignatureFunc(type)
 
       let response
       try {
         if (testFile.request) {
-          response = await testFile.request({ request, id, httpServerPort, endpoint, payload })
+          response = await testFile.request({ request, id, httpServerPort, endpoint, payload, signatureBuilder })
```

Comment:
I think instead of passing the `signatureBuilder`, you should just require it in the files that need it. Bitbucket doesn't need one of these right now anyway.

Reply:
Plus the implementations will be different for each